scripts/templates/helper.py

```diff
@@ -780,23 +780,30 @@ def make_details_lines(namespace, tags, obj):
         lines.append("")
         lines.append("@details")
 
-        for item in obj['details']:
-            if isinstance(item, dict):
-                for key, values in item.items():
-                    prologue = "    -"
-                    for line in split_line(subt(namespace, tags, key, True), 70):
-                        lines.append("%s %s"%(prologue, line))
-                        prologue = "     "
-                    for val in values:
-                        prologue = "        +"
-                        for line in split_line(subt(namespace, tags, val, True), 66):
+        # If obj['details'] is a list of bullet points, add the bullet point formatting to the lines.
+        if isinstance(obj['details'], list):
+            for item in obj['details']:
+                if isinstance(item, dict):
+                    for key, values in item.items():
+                        prologue = "    -"
+                        for line in split_line(subt(namespace, tags, key, True), 70):
                             lines.append("%s %s"%(prologue, line))
-                            prologue = "         "
-            else:
-                prologue = "    -"
-                for line in split_line(subt(namespace, tags, item, True), 70):
+                            prologue = "     "
+                        for val in values:
+                            prologue = "        +"
+                            for line in split_line(subt(namespace, tags, val, True), 66):
+                                lines.append("%s %s"%(prologue, line))
+                                prologue = "         "
+                else:
+                    prologue = "    -"
+                    for line in split_line(subt(namespace, tags, item, True), 70):
                         lines.append("%s %s"%(prologue, line))
                         prologue = "     "
+        # If obj['details'] is a string, then assume that it's already correctly formatted using markdown.
+        else:
```

Comment:
I feel like we should actually check `obj['details']` is an instance of a string and raise a `TypeError` when its not. That way we have a chance to explain to the user what's expected.

Reply:
That check is being done in the function __validate_details() on line 175. It was missing a couple of parentheses though which I have added in the latest commit.

Reply:
ah okay 👍